lib/internal/event_target.js

```diff
@@ -229,17 +229,17 @@ class EventTarget {
       throw new ERR_INVALID_ARG_TYPE('event', 'Event', event);
     }
 
-    if (this.#emitting.has(event.type) ||
+    if (this.#emitting.has(event) ||
         event[kTarget] !== undefined) {
```

Comment:
If we're only preventing the `Event` object from being dispatched again then the `this.#emitting` set can be dropped entirely here I think. The `event[kTarget] !== undefined` check is enough to prevent it from being dispatched recursively.

Reply:
Oh yeah, that makes sense. 

PTAL.